### PR TITLE
Update documentation and add examples

### LINKED ISSUE #5: Simplify device repo
#### Summary
As of now, its too complicated and error-prone. 

No. | Remove | Why
---|---|---
1 | Operating temperature | we're not copy-pasting datasheets
2 |RelativeHumidity | we're not copy-pasting datasheets
3 |Compliancy | we're not copy-pasting datasheets 

No. | Issues | Why
---|---|---
1 | Not everyone will install the validation tools | Lot of hassle to install
2 | yaml thing doesn't work well | it barely gives input (or I didn't install correctly). Either way, it needs improvements
3 | Not clear what to enter | list all options clearly as descriptions 
4 | HW version | not clear
5 | SF version | not clear
6 | firmwareVersions version or numeric | choose 1, chis confuses ppl
7 | Firmware Version profiles are most of the times similar | EU863-870 & AS923 may refer to the same file (might not be an issue)
8 | vendorProfileID | what is it and how to get it
9 | pen & oui | what is it, where to find it. Why is it necessary?

#### What is missing? What do you want to see?
- [ ] Improve CONTRIBUTING.md
- [ ] Clear example with all options people can fill out
- [ ] Clarify required and optional fields
- [ ] Not an option to have HW & SF version as we have it now. Reconsider this part

#### How do you propose to implement this?
I will work on documentation. Johan to provide feedback

#### Can you do this yourself and submit a Pull Request?
Yes, with input from @johanstokking

### LINKED ISSUE #9: Add documentation
#### Summary

Add documentation

#### Why do we need this?

For device makers to help them contributing to the device repository

#### What is already there? What do you see now?

JSON schema validation which helps in editors, and a few examples

#### What is missing? What do you want to see?

Full examples and an elaborate `README.md`

#### How do you propose to implement this?

Write and iterate via reviews with device makers

#### Can you do this yourself and submit a Pull Request?

Yes

### LINKED ISSUE #10: Validate examples as pull request status check
#### Summary

Validate payload encoder/decoder examples in the Validate status check for pull requests.

#### Why do we need this?

To make sure that payload codecs and examples work correctly.

#### What is already there? What do you see now?

Validation of the fields, and the ability to declare codecs and examples, but no validation of the examples.

#### What is missing? What do you want to see?

Validate the examples by running the codecs and assert the output.

#### How do you propose to implement this?

Add a sandbox environment which reflects what's used in The Things Stack.

#### Can you do this yourself and submit a Pull Request?

Yes

### PR DESCRIPTION
**Pull request by laurensslats**
#### Summary
Update CONTRIBUTING, added partner "Example" & Decentlab

Closes #5 
Closes #9 
Closes #10 

#### Changes
- Update CONTRIBUTING.md
- Added an "Example" company with products "X" and "Y"
- Added Decentlab vendor with 1 product

#### Notes for Reviewers
@johanstokking 
- Please first see my PRs in [LoRaWAN Schema](https://github.com/lorawan-schema/draft-devices/pull/3). This needs to be merged first to make GitHubs validation work.
- Please do a quick review. As a next step, I will ask someone within the company to add a demo device to see if all is clear. Next step is to distribute it to a small group of partners.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,49 +1,224 @@
...
 
 When contributing, make sure the models are validated against the schema and that all files are formatted correctly. This repository contains tooling to help you with that.
 
+See folder lorawan-devices/vendor/example to see a full overview of a
+
 ## Prerequisites
 
-[Download and install Node.js and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm). To check your Node.js and npm version:
+Node.js version 12.6+
+npm version 6.12+
```
Comment (johanstokking) [inline CONTRIBUTING.md:12]:
```suggestion
- Node.js version 12 or higher
- npm version 6 or higher
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,49 +1,224 @@
...
+[Visual Studio Code](https://code.visualstudio.com/) is a great editor for editing schemas. You can validate the models automatically using the [YAML plugin](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml).
 
-```bash
-$ npm run validate
+The YAML plugin supports you with filling out the document. When clicking ctrl + space, all options are shown. The Debug Console of Visual Studio also gives you feedback by highlighting the incorrect fields.
+
+## File structure
+
+```
+lorawan-device
```
Comment (johanstokking) [inline CONTRIBUTING.md:37]: Remove top-level dir

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,49 +1,224 @@
...
+│   │   ├── product-profile1.yaml
+│   │   ├── product-profile2.yaml
+│   │   ├── company-logo.svg
 ```
 
-### Visual Studio Code
+## File format
 
-[Visual Studio Code](https://code.visualstudio.com/) is a great editor for editing schemas. You can validate the models automatically using the [YAML plugin](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml).
+LoRaWAN Devices are defined in YAML files. Most settings are optional, when not specifying optional settings, the LoRaWAN industry standard will be used.
```
Comment (johanstokking) [inline CONTRIBUTING.md:55]: > the LoRaWAN industry standard will be used

What is this?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,49 +1,224 @@
...
+│   ├── index.yaml
+
+```
+
+```yaml
+vendors:
+  - id: company-x
+    name: Company X with devices Y and Z
+    pen: 99 #optional field, find your pen here: https://www.iana.org/assignments/enterprise-numbers/enterprise-numbers
+    oui: FCD6BD #optional field, find oui overview here; http://standards-oui.ieee.org/oui.txt
```
Comment (johanstokking) [inline CONTRIBUTING.md:74]:
```suggestion
    pen: 99 # IANA Private Enterprise Number (optional, see https://www.iana.org/assignments/enterprise-numbers/enterprise-numbers)
    oui: FCD6BD # IEEE OUI (optional, see http://standards-oui.ieee.org/oui.txt)
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,49 +1,224 @@
...
+    website: https://www.company-x.com
+    logo: example/logo.svg
+  - id: company-y
+    name: Company Y with devices X and Z
+    website: https://www.company-y.com
+    logo: example/logo.png #use a svg or png file
```
Comment (johanstokking) [inline CONTRIBUTING.md:80]:
```suggestion
    logo: example/logo.png # Local SVG or PNG file
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,49 +1,224 @@
...
+```
+
+
+#### End Device overview per vendor
```
Comment (johanstokking) [inline CONTRIBUTING.md:84]: Title Case

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,49 +1,224 @@
...
+Create a folder with your company name and add an index.yaml file to list all your devices:
```
Comment (johanstokking) [inline CONTRIBUTING.md:85]:
```suggestion
Create a folder with your company name and add `index.yaml` file to list all your devices:
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,49 +1,224 @@
...
+
+```yaml
+endDevices:
+  - product-x
+  - product-y
+```
 
-Put this in your `.vscode/settings.json` settings:
+#### Define each product
+Create a .ymal file for each product
```
Comment (johanstokking) [inline CONTRIBUTING.md:99]:
```suggestion
Create a `.yaml` file for each product
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,49 +1,224 @@
...
+
+```
+company_name
+├── product-x-info.yaml
```
Comment (johanstokking) [inline CONTRIBUTING.md:103]:
```suggestion
├── product-x.yaml
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,49 +1,224 @@
...
+  relativeHumidity:
+    min: 0
+    max: 0.97
+ipCode: IP64 #IP rating of your device
+keyProvisioning:
+  - custom
+  - join server
+keySecurity: none
+productURL: https://www.bosch-connectivity.com/products/connected-mobility/parking-lot-sensor/
+dataSheetURL: https://www.bosch-connectivity.com/media/product_detail_pls/parking-lot-sensor-datasheet.pdf
```
Comment (johanstokking) [inline CONTRIBUTING.md:167]: Remove Bosch reference

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,49 +1,224 @@
...
+    numeric: 2
+    hardwareVersions:
+      - '2'
+    profiles:
+      EU863-870:
+        id: product-profile-02
+        lorawanCertified: true
+      AS923:
+        id: product-profile-02
+sensors: #list all sensors. Enter ctrl + space to find all values. Options include: temperature,humidity,pir,light,accelerometer,gyroscope,magnetometer,gps,button,auxiliary,battery,rssi,snr,link,sound,co2, hvoc,proximity,water,ph,barometer,moisture,distance,dust,vibration
```
Comment (johanstokking) [inline CONTRIBUTING.md:143]: Remove options

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,49 +1,224 @@
...
+#### Define each product
+Create a .ymal file for each product
+
+```
+company_name
+├── product-x-info.yaml
+```
+
+```yaml
+id: product-x #identifier to correspond with endDevices ID in index.yaml
```
Comment (johanstokking) [inline CONTRIBUTING.md:107]:
```suggestion
id: product-x # Identifier to correspond with endDevices ID in index.yaml
```

Apply everywhere

File: vendor/bosch/tps110-0233.yaml [hunk 13]
```diff
@@ -1,5 +1,5 @@
 id: tps110-0223
-vendorProfileID: 1
+vendorProfileID: 69
```
Comment (johanstokking) [inline vendor/bosch/tps110-0233.yaml:2]: Did they provide this?

File: vendor/decentlab/dl-iam-001-decoder.js [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+  }
+};
+
+function main() {
+  console.log(decentlab_decoder.decode("020bbd007f0b926a515d48bc4e0262006981c7000093d4000b0111"));
+  console.log(decentlab_decoder.decode("020bbd006f0b926a515d48bc4e02620069000b0111"));
+  console.log(decentlab_decoder.decode("020bbd00010b92"));
+}
+
+main();
```
Comment (johanstokking) [inline vendor/decentlab/dl-iam-001-decoder.js:114]: Remove

File: vendor/index.yaml [hunk 15]
```diff
@@ -1,7 +1,15 @@
 vendors:
+  - id: example
+    name: Example vendor with XYZ devices
+    website: https://www.example.com
+    logo: example/logo.svg
```
Comment (johanstokking) [inline vendor/index.yaml:5]: Remove from this file

File: vendor/index.yaml [hunk 16]
```diff
@@ -1,7 +1,15 @@
...
   - id: bosch
     name: Bosch Connected Devices and Solutions GmbH
     pen: 42
     oui: FCD6BD
     website: https://www.bosch-connectivity.com
     logo: bosch/logo.svg
+  - id: decentlab
+    name: Decentlab GmbH
+    website: https://www.decentlab.com/
+    logo: decentlab/decentlab-logo.svg
```
Comment (johanstokking) [inline vendor/index.yaml:15]: I'm sure they have an OUI, and probably also a PEN

File: vendor/bosch/tps110-0233.yaml (hunk 13 above)
Comment (laurensslats) [reply vendor/bosch/tps110-0233.yaml:2]: Took from the list you shared: https://github.com/TheThingsNetwork/lorawan-devices/issues/5#issuecomment-648782960

File: vendor/decentlab/dl-iam-001-decoder.js (hunk 14 above)
Comment (laurensslats) [reply vendor/decentlab/dl-iam-001-decoder.js:114]: Decoder might still not work, will check with Decentlab for the right file.

File: vendor/index.yaml (hunk 16 above)
Comment (laurensslats) [reply vendor/index.yaml:15]: Didn't find Decentlab in the files:
- http://standards-oui.ieee.org/oui.txt
- https://www.iana.org/assignments/enterprise-numbers/enterprise-numbers

**Comment by laurensslats** [comment]
> I'll split the codec from profile in the schema and make sure we can test the codecs as part of CI

I'd rather not split the profile page. Vendors already need to work on 4 different files, a 5th one would only make things more complicated. There are also not too many fields in the profile file, it won't cost much effort to create these files.

**Pull request merged**